File: src/main/kotlin/dev/jefy/connectpro/marketplace/applicaion/query/JobApplicationQuery.java
package dev.jefy.connectpro.marketplace.applicaion.query;

import org.jspecify.annotations.NullMarked;

import java.util.List;

import dev.jefy.connectpro.marketplace.applicaion.dtos.JobApplicationResponseForEmployer;
import dev.jefy.connectpro.marketplace.applicaion.dtos.JobApplicationResponseForUser;
import dev.jefy.connectpro.marketplace.domain.vo.JobApplicationId;
import dev.jefy.connectpro.portfolio.domain.vo.JobPostId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface JobApplicationQuery {
    JobApplicationResponseForUser getForUser(JobPostId jobPostId);
    JobApplicationResponseForEmployer getForEmployer(JobApplicationId applicationId);
    List<JobApplicationResponseForUser> findForCurrentUser();
    List<JobApplicationResponseForEmployer> findByJobPost(JobPostId jobPostId);
}

File: src/main/kotlin/dev/jefy/connectpro/marketplace/application/query/JobApplicationQuery.kt
package dev.jefy.connectpro.marketplace.application.query

import dev.jefy.connectpro.marketplace.application.dtos.JobApplicationResponseForEmployer
import dev.jefy.connectpro.marketplace.application.dtos.JobApplicationResponseForUser
import dev.jefy.connectpro.marketplace.domain.vo.JobApplicationId
import dev.jefy.connectpro.portfolio.domain.vo.JobPostId

/**
 * @author Jôph Yamba
 */
interface JobApplicationQuery {
    fun getForUser(jobPostId: JobPostId): JobApplicationResponseForUser
    fun getForEmployer(applicationId: JobApplicationId): JobApplicationResponseForEmployer
    fun findForCurrentUser(): List<JobApplicationResponseForUser>
    fun findByJobPost(jobPostId: JobPostId): List<JobApplicationResponseForEmployer>
}
